### DISCUSSION TITLE
Spirv-cross for shader translation

### DISCUSSION THREAD

**Discussion by Tastaturtaste** [gfx-rs/wgpu#2953] [Q&A] [answered]
Does wgpu still use spirv-cross for shader translation? [This article](https://gfx-rs.github.io/2021/12/25/this-year.html) suggests that since 0.10.0 naga replaced spirv-cross, but the [docs upto 0.12.0](https://docs.rs/wgpu/0.12.0/wgpu/enum.ShaderSource.html) still say that spirv-cross is used for shader translation. Are the docs wrong?

**Comment by grovesNL** [comment, marked as answer]
Hi! :wave: The docs are outdated - it should say `naga` instead of `spirv_cross` there (and remove the reference to `gfx`, since it's not used anymore).

We'd be happy to accept a PR changing the comment if you'd like to submit one!

**Reply by Tastaturtaste** [reply]
Thanks, that was a fast answer if I have ever seen one.

**Reply by Tastaturtaste** [reply]
> We'd be happy to accept a PR changing the comment if you'd like to submit one!

The comment is correct in the latest release 0.13.*. Should I make pull requests for branches v0.10, v0.11 and v0.12?

**Reply by cwfitzgerald** [reply]
We don't do patch releases for old versions, so I wouldn't bother, thanks for the offer though.

**Comment by Tastaturtaste** [comment]
Another thing I just ran into is, that ShaderSource::Glsl is the only type in wgsl/lib.rs with `naga` types in it, despite not behind the 'naga' feature flag. Would it be possible to remove these from the public interface while the 'naga' feature flag is not enabled?
